apps/wallet/src/ui/app/redux/slices/sui-objects/index.ts

```diff
@@ -33,66 +34,96 @@ export const fetchAllOwnedAndRequiredObjects = createAsyncThunk<
-        const objectIDs = allObjectRefs
-            .filter((anObj) => {
-                const fetchedVersion = getObjectVersion(anObj);
-                const storedObj = suiObjectsAdapterSelectors.selectById(
-                    state,
-                    getObjectId(anObj)
+>(
+    'sui-objects/fetch-all',
+    async (_, { getState, extra: { api }, rejectWithValue }) => {
+        try {
+            const state = getState();
+            const {
+                account: { address },
+            } = state;
+            const allSuiObjects: SuiObject[] = [];
+            if (address) {
+                const allObjectRefs =
+                    await api.instance.fullNode.getObjectsOwnedByAddress(
+                        `${address}`
+                    );
+                const objectIDs = allObjectRefs
+                    .filter((anObj) => {
+                        const fetchedVersion = getObjectVersion(anObj);
+                        const storedObj = suiObjectsAdapterSelectors.selectById(
+                            state,
+                            getObjectId(anObj)
+                        );
+                        const storedVersion = storedObj
+                            ? getObjectVersion(storedObj.reference)
+                            : null;
+                        const objOutdated = fetchedVersion !== storedVersion;
+                        if (!objOutdated && storedObj) {
+                            allSuiObjects.push(storedObj);
+                        }
+                        return objOutdated;
+                    })
+                    .map((anObj) => anObj.objectId);
+                objectIDs.push(SUI_SYSTEM_STATE_OBJECT_ID);
+                const allObjRes = await api.instance.fullNode.getObjectBatch(
+                    objectIDs
                 );
-                const storedVersion = storedObj
-                    ? getObjectVersion(storedObj.reference)
-                    : null;
-                const objOutdated = fetchedVersion !== storedVersion;
-                if (!objOutdated && storedObj) {
-                    allSuiObjects.push(storedObj);
+                for (const objRes of allObjRes) {
+                    const suiObj = getObjectExistsResponse(objRes);
+                    if (suiObj) {
+                        allSuiObjects.push(suiObj);
+                    }
                 }
-                return objOutdated;
-            })
-            .map((anObj) => anObj.objectId);
-        objectIDs.push(SUI_SYSTEM_STATE_OBJECT_ID);
-        const allObjRes = await api.instance.fullNode.getObjectBatch(objectIDs);
-        for (const objRes of allObjRes) {
-            const suiObj = getObjectExistsResponse(objRes);
-            if (suiObj) {
-                allSuiObjects.push(suiObj);
             }
+            return allSuiObjects;
+        } catch (err) {
+            const error = err as Error;
+            reportSentryError(error);
```

Comment:
Since we are doing a generic logging of the error, we could move that to a middleware and then we won't have to include it for every action. Something like this example [here](https://redux.js.org/understanding/history-and-design/middleware#the-final-approach) (the `crashReporter`)